back/app/models.py

```diff
@@ -62,4 +66,75 @@ def from_dict(cls, data):
             return cls(name=name, age=age, level=level, msg=msg, important=important, status=status, _id=_id)
         except Exception as e:
             raise ValueError(f"Error creating Data object from dict: {e}")
+        
+    @staticmethod
+    def get_statistics():
+        all_records = db.data.find({})
+        records_len = len(list(all_records.clone()))
+        avg_age=0
+        level_distribution={
+            "easy": 0,
+            "normal": 0,
+            "hard": 0
+            }
+        level_distribution_total = 0
+        important_frequency = {True:0,False:0}
+        status_frequency = {"open":0,"closed":0}
+        common_words = defaultdict(int)
+
+        for record in all_records:
+            #calculate avg_age    
+            a = datetime.strptime(record['age'], "%Y-%m-%d").date()
+            b = datetime.now().date()
+            delta = b - a
+            avg_age += delta.days
+
+            #calculate level_distribution
+            level_distribution[record['level']] +=1
+            level_distribution_total +=1
+
+            #calculate important_frequency
+            important_frequency[record['important']] +=1
+
+            #calculate status_frequency
+            status_frequency[record['status']] +=1
+
+            #calculate common_words
+            message = str(record['msg'])
+            message.translate(str.maketrans('', '', string.punctuation)) #remove all punctuation
+            re.sub(' +', ' ', message) # remove multiple spaces e.g 'The   fox jumped   over    the log.'
+
+            for word in message.split():
+                common_words[word.title()] +=1
+
+        #safe gurad to prevent division by zero
+        records_len = max(records_len,1)
+        level_distribution_total = max(level_distribution_total,1)
+
+        #calculating percentages
+        avg_age *= 100/records_len
+        avg_age = round(avg_age,1)
+
+        level_distribution["easy"] *= (100 / level_distribution_total)
+        level_distribution["easy"] = round(level_distribution["easy"],1)
+
+        level_distribution["normal"] *= (100 / level_distribution_total)
+        level_distribution["normal"] = round(level_distribution["normal"],1)
+
+        level_distribution["hard"] *= (100 / level_distribution_total)
+        level_distribution["hard"] = round(level_distribution["hard"],1)
+
+
+        stat = {
+            "avg_age" : avg_age,
+            "level_distribution": level_distribution,
+            "important_frequency": important_frequency,
+            "status_frequency" : status_frequency,
+            "common_words": list(common_words.items())
```

Comment:
Not all words are common_words. please select only the top %20 most common words in the whole database

also make sure to return a List of tuples 
```py
common_words_in_msg = [
    ("break",3),
    ("new", 5)
]
```
**not!!** 
`common_words_in_msg = [
    ["break",3],
    ["new", 5]
]`